agent/api/task_unix.go

```diff
@@ -42,14 +44,34 @@ const (
 )
 
 // platformFields consists of fields specific to Linux for a task
-type platformFields struct {}
+type platformFields struct{}
 
 func (task *Task) adjustForPlatform(cfg *config.Config) {
 	task.lock.Lock()
 	defer task.lock.Unlock()
 	task.MemoryCPULimitsEnabled = cfg.TaskCPUMemLimit.Enabled()
 }
 
+func (task *Task) initializeCgroupResource(cgroupPath string, resourceFields *taskresource.ResourceFields) error {
```

Comment:
`initializeCgroupResource` makes it seem like we are actually initializing cgroup here. Are we actually doing that? If we are, this method shouldn't be called in `PostUnmarshalTask`. If we are not, can you please rename this to `injectCgroupResource` or something that conveys that meaning?

Reply:
This method just initializes the `CgroupResource` object. In`progressTask`, actual creation of cgroup happens by calling `Create()` on `CgroupResource` during resource state transitions. I am not sure if `inject` conveys the right meaning.

Reply:
How about `initCgroupResourceSpec()`?